### ISSUE TITLE
Reedline appears to be swallowing Cmd Key on a mac

### ISSUE THREAD

**Issue by stormasm** [nushell/reedline#420]
@sholderbach sorry to be back here again with a similar issue...

@jntrnr and I were doing some testing this evening with the following example in the crossterm source code on the default mac terminal...

https://github.com/crossterm-rs/crossterm/blob/master/examples/event-read.rs

Step 1 is to comment out this line of code first

line 76
```rust
    execute!(stdout, DisableMouseCapture)?;
```

When you run this example it does 2 things...
1) clears the screen
2) clears out the scroll back
on a default mac terminal when you hit "Command"-k

JT and I both were able to see this work...
Maybe @fdncred might try it too :)

Meaning this example shows that although it doesn't **report the event**

it also doesn't swallow the key sequence but lets it go through to the
mac terminal which does its job of clear screen plus scroll back

So reedline maybe swallows the CMD-k key sequence because no event is reported ?

If it let that key sequence go through just like the above example the Command
sequences on macs would work ?

**Comment by stormasm** [comment]
I looked through the rustyline source code and did not spot any code that deals with 

https://en.wikipedia.org/wiki/Command_key
U+2318

So I am speculating that Rustyline just let the command sequence go through also and thats why in nushell 0.44 the Command-K sequence worked...

**Comment by fdncred** [comment]
`cmd+k` works for me but only with mac's terminal.app. `cmd+k` also works in nushell in mac's terminal.app already for me.

**Comment by stormasm** [comment]
if you see this behavior.
and I am sure you will..
I guess this behavior makes perfect sense...
because reedline has no idea what just happened.
and it never will because it can't read cmd-k
(for all of the reasons we have been discussing for the past day)
so it puts you back exactly where you were before you hit cmd-k
and that (I believe) is the end of the story 😄

**Comment by sholderbach** [comment]
Yeah that sounds like my understanding as well. I wrote my rambling after trying to find potential API solutions for Mac on Sunday in the first point under https://github.com/nushell/reedline/issues/120#issuecomment-1114331332 So really solving the Terminal.app quirk via keybindings is out of the question (without building a system level keylogger :P) the only change I could imagine to solve that is completely changing the drawing strategy (if we don't find good inspiration from other line editors, a hard no as the current solution works but is hard to get right with resizing/not scrolling at the bottom of the screen etc.).
